src/main/java/repository/BookRepository.java

```diff
@@ -0,0 +1,152 @@
+package repository;
+
+import model.book.Book;
+import model.book.impl.PaperBook;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Has the role of containing all the books in the library and execute simple operations like:
+ * - adding book;
+ * - getting all books
+ * - etc.
+ */
+public class BookRepository {
+    private final Map<String, Book> books = new HashMap<>();
+    private Map<String, Map<String, LocalDate[]>> usersWithBorrowedBooks = new HashMap<>();
+    private Map<String, List<String>> offeredBooks = new HashMap<>();
+
+    public BookRepository() {
+    }
+
+    /**
+     * Provides checks for the validity of the claim - user has offered books, user has exact offered book,
+     * and returns the corresponding message.
+     *
+     * @param username Unique identifier of the user.
+     * @param ISBN     Unique identifier of the book.
+     * @return Message explayning the actions taken/not taken.
+     */
+    public String borrowBook(String username, String ISBN) {
+        if (offeredBooks.containsKey(username)) {
+            if (offeredBooks.get(username).contains(ISBN)) {
+                registerBorrowedBook(username, ISBN);
+                offeredBooks.get(username).remove(ISBN);
+                ((PaperBook)books.get(ISBN)).getUsersInQueue().remove(username);
```

Comment:
better to use a method to remove users from queue instead of getting the whole queue and using it's methods